lib/boat_tracker.rb

```diff
@@ -0,0 +1,34 @@
+class BoatTracker
+
+  def self.any_boat_hit?(boat_list, target_coords)
+    hit = false
+    boat_list.each do |boat|
+      boat.coords_hash.each_key do |boat_coords|
+        if boat_coords == target_coords
+          hit = true
+        end
+      end
+    end
+    hit
+  end
+
+  def self.which_boat_hit?(boat_list, target_coords)
+    attacked_boat = nil
+    boat_list.each do |boat|
```

Comment:
Nested looping is a code smell, maybe you could extract the inner loop into another method?